### DISCUSSION TITLE
how to run termux background automatically when android device reboot

### DISCUSSION THREAD

**Discussion by Jzhenli** [termux/termux-app#3470] [Q&A] [answered]
Hi, if it's possible for the termux  app auto run background when the android device reboot? 

Thanks

**Comment by agnostic-apollo** [comment, marked as answer]
You can use termux-boot to run a `RUN_COMMAND` intent to start a bash shell or something with `RUN_COMMAND_SESSION_ACTION` as `2` or `3`.

 https://github.com/termux/termux-boot/

https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent

https://github.com/termux/termux-app/blob/v0.118.0/termux-shared/src/main/java/com/termux/shared/termux/TermuxConstants.java#L943

**Comment by risharde** [comment]
@agnostic-apollo are you sure this is supposed to run termux in the background (Android 7)? I don't see any termux processes running when I use ps on boot
I performed:
ps | grep termux

**Reply by agnostic-apollo** [reply]
Where are you running ps?

And boot event is received ~60s after unlock screen on boot. You should see the standard termux notification if boot script and run command was successful.

Make sure auto start, etc in your android device settings is enabled for both apps. https://dontkillmyapp.com/

See also https://github.com/termux/termux-packages/blob/site/site/pages/en/projects/docs/repos/main/packages/termux-tools/logcat/index.md

**Reply by risharde** [reply]
I tried running ps after using 'adb shell'
I think I did this right after boot (did not wait the 60 seconds so I'll test this - thanks for letting me know about this!)
This is a Firestick so it might be a little more complicated to enable any settings.
I'll also look at the logcat resource
Thank you for the quick response, I'll work at it and report back!
